config/env/env_types.go

```diff
@@ -0,0 +1,161 @@
+package env
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strconv"
+)
+
+var TypeError = errors.New("type could not be converted")
```

Comment:
This error could communicate more to help users debug.  i.e. "expected type bool, received type 'foo' with value of 'bar'"